decidim-generators/lib/decidim/generators/app_generator.rb

```diff
@@ -333,6 +333,26 @@ def decidim_initializer
                   "# config.available_locales = Rails.application.secrets.decidim[:available_locales].presence || [:en]"
       end
 
+      def dev_performance_config
+        gsub_file "config/environments/development.rb", /^end\n$/, <<~CONFIG
+
+            # Performance configs for local testing
+            if ENV.fetch("RAILS_BOOST_PERFORMANCE", false).to_s == "true"
+              # Indicate boost performance mode
+              config.boost_performance = true
+              # Enable caching and eager load
+              config.eager_load = true
+              config.cache_classes = true
+              # Logging
+              config.log_level = :info
+              config.action_view.logger = nil
+              # Compress the HTML responses with gzip
+              config.middleware.use Rack::Deflater
+            end
+          end
+        CONFIG
+      end
```

Comment:
I am not sure i would like to ship this to the end user. Maybe something like this would be more appropriate?!

```suggestion
      def dev_performance_config
        return unless ENV.fetch("RAILS_BOOST_PERFORMANCE", false).to_s == "true"
        
        gsub_file "config/environments/development.rb", /^end\n$/, <<~CONFIG

            # Performance configs for local testing
            # Indicate boost performance mode
            config.boost_performance = true
            # Enable caching and eager load
            config.eager_load = true
            config.cache_classes = true
            # Logging
            config.log_level = :info
            config.action_view.logger = nil
            # Compress the HTML responses with gzip
            config.middleware.use Rack::Deflater
          end
        CONFIG
      end
```

Reply:
I actually thought this would be useful for the normal development app as well, so that we could also locally test this out easily without having to spend a lot of time configuring it. E.g. if we see a problem at the Lighthouse metrics, we can easily replicate the issue locally using this config. And also do occasional testing on pages that are not covered by the workflow itself.

Another thing is that we should not rely on the environment variables to decide what the generator does. We could introduce options to it and pass those options to the generator command but that would also mean that we have to modify the rake task for generating the application and add some parameters to it which requires more work. The introduced environment variable is meant to be an application runtime configuration option.

So in my opinion, we should ship these configs with the default app. It does not do anything by default, unless you specifically enable the performance configs. We are already doing similar stuff for the production environment config (within the same generator) which is why I took this approach as I thought that would be the preferred way to go.